### DISCUSSION TITLE
dispatch job on specific connection

### DISCUSSION THREAD

**Discussion by slimani-dev** [spatie/laravel-webhook-client#119] [Ideas]
# The problem 

I have few jobs in my application that Run in the Background in a database queue connection, the problem is that when i started using the database connection i lost the HTTP request information of the Webhook Job and i need the request headers 

# The requested feature 
it would be amazing if there is an option to choose the queue connection for the Webhook job

something like this 
```php
namespace Spatie\WebhookClient;
...

class WebhookProcessor
{
    protected Request $request;

    protected WebhookConfig $config;
    
    ...

   protected function processWebhook(WebhookCall $webhookCall): void
    {
        try {
            $job = new $this->config->processWebhookJobClass($webhookCall);

            $webhookCall->clearException();

            dispatch($job)->onConnection('sync');

        } catch (Exception $exception) {
            $webhookCall->saveException($exception);

            throw $exception;
        }
    }

    ...

}

```

in line 62 of WebhookProcessor.php file

```php
dispatch($job)->onConnection('sync');
```
this solution is working for me but it means i will have to edit the files every time 

thank you

**Comment by freekmurze** [comment]
Feel free to send a PR with tests for this!

**Comment by wit3** [comment]
@moh-slimani define custom job processor ( that implement `\Spatie\WebhookClient\Jobs\ProcessWebhookJob` ) and ridefine constructor like this didn't work?

```php
class ProcessCustomJob extends ProcessWebhookJob implements ShouldQueue
{
    public function __construct(
        public WebhookCall $webhookCall
    ) {
        //$this->onQueue('custom_queue'); //this work for me
        $this->onConnection('sync'); // try this
    }

    /**
     * Execute the job.
     *
     * @return void
     */
    public function handle()
    {
    
    }
```
